lib/network/rest/utils.js

```diff
@@ -0,0 +1,13 @@
+'use strict'; // eslint-disable-line
+
+module.exports.explodePath = function explodePath(path) {
+    const pathMatch = /^(\/[a-zA-Z0-9]+)(\/([0-9a-f]*))?$/.exec(path);
+    if (pathMatch) {
+        return {
+            service: pathMatch[1],
+            key: (pathMatch[3] !== undefined && pathMatch[3].length > 0 ?
+                  pathMatch[3] : undefined),
+        };
+    }
+    return undefined;
```

Comment:
Ultimately, throw would end being try..catch 'd. Just returning the error directly when we know it's an error would make sense.